api/csi.go

```diff
@@ -110,11 +107,17 @@ type CSIVolume struct {
 	Parameters     map[string]string       `hcl:"parameters"`
 	Context        map[string]string       `hcl:"context"`
 
-	// Allocations, tracking claim status
-	ReadAllocs  map[string]*Allocation
+	// ReadAllocs is a map of allocation IDs for tracking reader claim status.
+	// The Allocation value will always be nil; clients can populate this data
+	// by iterating over the Allocations field.
+	ReadAllocs map[string]*Allocation
+
+	// WriteAllocs is a map of allocation IDs for tracking writer claim
+	// status. The Allocation value will always be nil; clients can populate
+	// this data by iterating over the Allocations field.
 	WriteAllocs map[string]*Allocation
```

Comment:
Curious why we chose this representation?  If the map value is never a populated alloc, how does it compare to using `map[string]bool` (explicitly meaningless value) or a plain slice `[]string` (avoid value but losing uniqueness)? Not advocating for changing it, specially if we are constrained by compatibility.

Reply:
It's solely constrained by compatibility at this point, unfortunately.

With 20/20 hindsight, we would have had the `nomad/structs` version store only the `CSIVolumeClaim`s and not allocations (which has been a huge complication with respect to garbage collected jobs and allocations), and then had this API struct return `WriteAllocs map[string]*AllocationListStub` and `ReadAllocs map[string]*AllocationListStub`, without an `Allocations` field at all. See https://github.com/hashicorp/nomad/issues/8734 for more